### ISSUE TITLE
Add functionResponseTypes to SQS's onEvent args

### ISSUE THREAD

**Issue by git-chrisp** [pulumi/pulumi-aws#2048]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

In order to return `batchItemFailures` in a Lambda assigned to a SQS queue, `functionResponseTypes` needs to be configured in the EventSourceMapping. That is currently not possible in the Node.js SDK. It needs to be added to the QueueEventSubscriptionArgs.

### Affected area/feature

https://github.com/pulumi/pulumi-aws/blob/a0b26b1cff4f5d7879e54f7771abff59446a0bb0/sdk/nodejs/sqs/sqsMixins.ts#L65

**Comment by EvanBoyle** [comment]
Just ran into this. This is kind of an important parameter. If your lambda returns without this parameter, all messages in the batch are deleted from SQS and will not be reprocessed. 

Reprocessing some messages from a batch is a critical use case!

**Comment by EvanBoyle** [comment]
In addition, the type of the `onEvent` lambda doesn't allow returning the necessary data `{batchItemFailures: [...])`

**Comment by EvanBoyle** [comment]
For anyone who runs into this, it seems that the cleanest workaround is to set `QueueEventSubscriptionArgs.batchSize` to 1 and then throw an error within the lambda if you want to process a particular message again. Obviously not ideal if you are trying to process high volumes of traffic.

**Comment by fmaerkl-sw** [comment]
No, the cleanest workaround without compromises in functionality is to just not use the QueueEventSubscription abstraction at all for now and instead implement the low-level resources directly, e.g.:
```typescript
new aws.lambda.Permission("queue-handler-permission", {
	action: "lambda:*",
	function: queueHandlerFunction,
	principal: "sqs.amazonaws.com",
	sourceArn: queue.arn,
});

new aws.lambda.EventSourceMapping("queue-handler", {
	enabled: true,
	eventSourceArn: queue.arn,
	functionName: queueHandlerFunction.name,
	functionResponseTypes: ["ReportBatchItemFailures"]
});
```

The QueueEventSubscription implementation can be used as a reference:
https://github.com/pulumi/pulumi-aws/blob/c02b3d3d8bdb18a591255bd1b0f3c160bb295dce/sdk/nodejs/sqs/sqsMixins.ts#L73-L105 

This is also the place where this issue could be fixed.
